packages/js/src/core/sign-in.ts

```diff
@@ -0,0 +1,47 @@
+import qs from 'query-string';
+
+import { withDefaultScopes } from '../utils/scope';
+
+export type SignInUriParameters = {
+  authorizationEndpoint: string;
+  clientId: string;
+  redirectUri: string;
+  codeChallenge: string;
+  state: string;
+  scope?: string | string[];
+  resource?: string[];
+};
+
+enum ResponseType {
+  code = 'code',
+}
+
+export const generateSignInUri = ({
+  authorizationEndpoint,
+  clientId,
+  redirectUri,
+  codeChallenge,
+  state,
+  scope,
+  resource,
+}: SignInUriParameters) => {
+  const queryStringWithoutResource = qs.stringify({
+    client_id: clientId,
+    redirect_uri: [redirectUri],
+    code_challenge: codeChallenge,
+    code_challenge_method: 'S256',
+    state,
+    response_type: ResponseType.code,
+    prompt: 'consent',
+    scope: withDefaultScopes(scope),
+  });
+
+  const queryStringWithOnlyResource = (resource ?? []).reduce(
```

Comment:
`resource?.reduce`?

Reply:
can not
![image](https://user-images.githubusercontent.com/10594507/151653354-965bbb7f-bd14-484c-825d-adb2e13d05a8.png)

Reply:
OK, I missed that one. So you will have to either fallback the array (`resource ?? []`) or the result string (`?? ''`). Then I'm OK with this.